reports/ingress-weighting/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	// "github.com/ministryofjustice/cloud-platform-how-out-of-date-are-we/reports/pkg/authenticate"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// resourceMap is used to store both string:string and string:map[string]string key
+// value pairs. The HOODAW API requires the first entry of map to contain a string:string,
+// the rest of the map consists of a primary key (string) with a value containing a map (string:string)
+// of Kubernetes namespaces and hostnames that don't contain the annotation contained within
+// the variable 'annotation'.
+type resourceMap map[string]interface{}
+
+var (
+	hoodawApiKey   = flag.String("hoodawAPIKey", os.Getenv("HOODAW_API_KEY"), "API key to post data to the 'How out of date are we' API")
+	hoodawHost     = flag.String("hoodawHost", os.Getenv("HOODAW_HOST"), "Hostname of the 'How out of date are we' API")
+	bucket         = flag.String("bucket", os.Getenv("KUBECONFIG_S3_BUCKET"), "AWS S3 bucket for kubeconfig")
+	configFile     = flag.String("configFile", "live-1-only", "Name of kubeconfig file in S3 bucket")
```

Comment:
@jasonBirchall , why do we need a seperate kubeconfig which has only-live-1. Can we not use the existing kubeconfig which concourse has?

Reply:
It's a great question. The k8s client package isn't brilliant at handling multiple contexts, it adds so much more complexity to the code. I felt this was the easiest win for all.